scene/main/viewport.cpp

```diff
@@ -929,6 +954,14 @@ Rect2 Viewport::get_visible_rect() const {
 	return r;
 }
 
+void Viewport::canvas_parent_mark_dirty(Node *p_node) {
+	bool request_update = gui.canvas_parents_with_dirty_order.is_empty();
+	gui.canvas_parents_with_dirty_order.insert(p_node->get_instance_id());
+	if (request_update) {
+		MessageQueue::get_singleton()->push_callable(callable_mp(this, &Viewport::_process_dirty_canvas_parent_orders));
```

Comment:
Just a note, something I noticed on my own PRs:

This looks like it will get flushed every time the `message_queue` is flushed, so it might mean it gets repeatedly done on several physics ticks and again before a frame (if the same parent node gets hit repeatedly on each tick). I originally did a flush on each physics tick and before a frame (for safety), but on looking through the actual uses made for the NOTIFICATION (principally draw order for canvas items, and things concerned with rendering), I realised it may be possible to flush this only once before a frame.

This isn't necessary to start with but might be worth trying in a later PR. In my tests so far I've had no problems from flushing just once before a frame.